ArrayIndexOutOfBoundsException when non alphanumeric String for camelCase method

The Strings.camelCase method is throwing a ArrayIndexOutOfBoundsException when it receives a non alphanumeric String like for example `__`:

For reproducing, look this [#464](https://github.com/halkyonio/primaza-poc/issues/464)

Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
    at io.sundr.utils.Strings.camelCase (Strings.java:258)
    at io.sundr.builder.internal.checks.DublicatePropertyCheck.visit (DublicatePropertyCheck.java:19)
    at io.sundr.builder.internal.checks.DublicatePropertyCheck.visit (DublicatePropertyCheck.java:11)
    at io.sundr.builder.Visitor.visit (Visitor.java:30)
    at io.sundr.builder.VisitorWiretap.visit (VisitorWiretap.java:79)
    at io.sundr.builder.Visitable.accept (Visitable.java:67)
    at io.sundr.builder.Visitable.lambda$accept$1 (Visitable.java:81)
    at java.util.Optional.ifPresent (Optional.java:178)
    at io.sundr.builder.Visitable.accept (Visitable.java:73)
    at io.sundr.builder.Visitable.accept (Visitable.java:53)
    at io.sundr.builder.Visitable.accept (Visitable.java:30)
    at io.sundr.builder.internal.processor.BuildableProcessor.process (BuildableProcessor.java:82)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor (JavacProcessingEnvironment.java:1021)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs (JavacProcessingEnvironment.java:937)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run (JavacProcessingEnvironment.java:1265)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing (JavacProcessingEnvironment.java:1380)
    at com.sun.tools.javac.main.JavaCompiler.processAnnotations (JavaCompiler.java:1272)
    at com.sun.tools.javac.main.JavaCompiler.compile (JavaCompiler.java:946)
    at com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0 (JavacTaskImpl.java:104)
    at com.sun.tools.javac.api.JavacTaskImpl.invocationHelper (JavacTaskImpl.java:152)
    at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:100)
    at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:94)
    at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:136)
    at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:183)
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1140)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:193)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:299)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:193)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:106)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:963)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:296)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:199)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/core/src/main/java/io/sundr/utils/Strings.java
+++ b/core/src/main/java/io/sundr/utils/Strings.java
@@ -248,18 +248,52 @@ public final class Strings {
   }
 
   /**
-   * Converts the String into a camel case String.
+   * Converts the given string to camelCase format.
+   * <p>
+   * This method takes a string input and converts it to camelCase format,
+   * where each word in the input string is capitalized (except the first word)
+   * and concatenated together without spaces. Non-alphanumeric characters are treated as word separators.
+   * <p>
+   * If the input string is {@code null}, {@code null} is returned.
+   * If the input string is empty or contains only non-alphanumeric characters, an empty string is returned.
+   * <p>
+   * For example:
    *
-   * @param name The name.
-   * @return The name in camel case.
+   * <pre>{@code
+   * camelCase("helloWorld") returns "helloWorld"
+   * camelCase("hello_world") returns "helloWorld"
+   * camelCase("hello world") returns "helloWorld"
+   * camelCase("foo-bar-baz") returns "fooBarBaz"
+   * camelCase("123_456_789") returns "123456789"
+   * camelCase("   ") returns ""
+   * camelCase(null) returns null
+   * }</pre>
+   *
+   * @param name the string to convert to camelCase
+   * @return the input string converted to camelCase format, or {@code null} if the input string is {@code null},
+   *         or an empty string if the input string is empty or contains only non-alphanumeric characters
    */
   public static final String camelCase(String name) {
+    if (name == null) {
+      return null;
+    } else if (name.isEmpty() || name.trim().isEmpty()) {
+      return "";
+    }
     String[] parts = name.split("[^a-zA-Z0-9]");
-    return parts[0] + Stream.of(parts)
-        .skip(1)
-        .filter(s -> s != null && s.length() > 0)
-        .map(v -> Character.toUpperCase(v.charAt(0)) + v.substring(1))
-        .collect(Collectors.joining());
+    if (parts.length == 0) {
+      return null;
+    }
+    StringBuilder builder = new StringBuilder();
+    for (int i = 0; i < parts.length; i++) {
+      String word = parts[i];
+      if (i == 0 || builder.toString().isEmpty()) {
+        word = word.isEmpty() ? word : word.substring(0, 1).toLowerCase() + word.substring(1);
+      } else {
+        word = word.isEmpty() ? word : Character.toUpperCase(word.charAt(0)) + word.substring(1).toLowerCase();
+      }
+      builder.append(word);
+    }
+    return builder.toString();
   }
 
   /**

--- a/core/src/test/java/io/sundr/utils/StringsTest.java
+++ b/core/src/test/java/io/sundr/utils/StringsTest.java
@@ -50,4 +50,22 @@ public class StringsTest {
     assertEquals("$_Dog", capitalizeFirst("$_dog"));
     assertEquals("1Dog", capitalizeFirst("1dog"));
   }
+
+  @Test
+  public void testCamelCase() {
+    assertEquals(null, camelCase(null));
+    assertEquals("", camelCase(""));
+    assertEquals("", camelCase("    "));
+    assertEquals("helloWorld", camelCase("hello_world"));
+    assertEquals("helloWorld", camelCase("hello world"));
+    assertEquals("helloWorld", camelCase("Hello World"));
+    assertEquals("fooBarBaz", camelCase("foo-bar-baz"));
+    assertEquals("123456789", camelCase("123_456_789"));
+    assertEquals("123", camelCase("123"));
+    assertEquals("dog", camelCase("_dog"));
+    assertEquals("fooBar", camelCase("FooBar"));
+    assertEquals("foobar", camelCase("foobar"));
+    assertEquals("fooBar", camelCase("fooBar"));
+    assertEquals(null, camelCase("__"));
+  }
 }

